### ISSUE TITLE
Unnecessary panic which causes a double panic.

### ISSUE THREAD

**Issue by Multirious** [DarrenTsung/rust-pretty-assertions-sorted#2]
at `pretty_assertions_sorted-1.2.0/src/lib.rs:101`. Debug print should not be able to returns error. Thanks [Finomnis](https://stackoverflow.com/a/73624390/14718394) that found the problem.

The reproducible example using `serde_json`:
```rust
#[cfg(test)]
mod test {
    use pretty_assertions_sorted::assert_eq_sorted;
    use serde_json::json;

    #[test]
    fn double_panic() {
        // Should not double panic
        assert_eq_sorted!(json!({"a":0}), json!(null));
    }
}
```
bro guiding me good

**Comment by Finomnis** [comment]
Should not be able to *panic*. Returning errors is fine-ish. I'd still try to handle it gracefully in this case, like printing a message telling the user that this struct cannot be sorted.

@Multirious A bug report should at least contain a minimal reproducible example, what it should do, and what it does. Please insert the crashing `json` example here.

**Comment by DarrenTsung** [comment]
Thanks for the bug report! I didn't catch that this case would cause a double-panic, definitely not intended. This is fixed in https://github.com/DarrenTsung/rust-pretty-assertions-sorted/commit/d5892e22263e74fc696013f89aee3ee96a3235b4 and published as `1.2.1`.

Re: panic being bad style - Unfortunately the panic in the Debug representation is necessary in order to leverage the `pretty_assertions` crate as the `pretty_assertions::Comparison` only has an API for passing in objects that are `Debug` and not the raw strings.. actually scratch that, maybe passing in the Debug representation rendered as a string would work.

Either way panicking in a test is expected behavior here - the alternative would be to fallback to the unsorted debug representation and I think that should be the user's action instead of the crate automatically doing so (logging something doesn't seem enough because logs are suppressed for passing tests).
